docs/timeout.rst

```diff
@@ -0,0 +1,38 @@
+.. _timeout:
+
+Setting the Lock Timeout
+========================
+
+Use `pglock.timeout` to dynamically set Postgres's ``lock_timeout`` variable.
+This setting is isolated to the thread executing code.
+
+For example, below we've ensured that no statement will take longer
+than three seconds to acquire a lock:
+
+.. code-block:: python
+
+    import pglock
+
+    with pglock.timeout(seconds=3):
+        # If any SQL statement takes longer than 3 seconds to acquire a lock,
+        # a django.db.utils.OperationalError will be raised.
+
+.. tip::
+
+    `pglock.timeout` takes the same arguments as Python's ``timedelta``
+    object, meaning you can also pass ``milliseconds``. You can also supply
+    a ``timedelta`` object as the first argument.
+
+`pglock.timeout` can be nested like so:
+
+.. code-block:: python
+
+    with pglock.timeout(seconds=2):
+        # Every statment here will have a lock timeout of 2 seconds
+
+        with pglock.timeout(seconds=5):
+            # Every statement will now have a lock timeout of 5 seconds
+
```

Comment:
In some contexts this may be slightly confusing. For example, suppose someone wrapped an entire view in a pglock.timeout(seconds=2) manager with the deliberate intention that nothing should ever wait for more than 2 seconds. Someone could then easily end up adding some pglock.timeout(seconds=10) wrapper around a piece of code which is hit by the view but is no were "near" the orginal wrapper (e.g. in a signal or something). To prevent that, we could have something like an `allow_nesting` kwarg, which would be passed in as False to pglock.timeout(seconds=2, allow_nested=False) in this scenario. Sort of like the reverse of Django atomic's 'durable' kwarg.

Reply:
I would say that's exactly the behaviour one would want from the `allow_nested=False` part, but no need to do it just now anyway.